spec/flows/early_result_spec.rb

```diff
@@ -109,6 +109,18 @@
         expect(page).to have_content "Your client's key eligibility totals"
       end
 
+      it "when I go straight to results and use the back button it shows the banner" do
+        confirm_screen("outgoings")
+        outgoings_url = current_path
+        expect(page).to have_content("Gross monthly income limit exceeded")
+        click_on "Go to results page"
+        expect(page).to have_current_path(/\A\/check-result/)
+        expect(page).to have_content "Your client's key eligibility totals"
+        visit outgoings_url # simulates using the back button to return to outgoings
```

Comment:
I see why you've done this - I wonder if we can add a test that really checks the back button is working as expected? Perhaps an end2end test might work? But not too sure

Reply:
I did spend some time trying to get the back button to work, essentially as we move from ` def early_result_redirect` to `def show` in `results_controller.rb` we lose the ` @previous_step = params[:step].to_sym` value. This doesnt matter to the back button as the browser handles it and so does our back link.

To get the test to work we would need to add code that would only be there for the test to work, which is bad coding.

Having looked at the other uses of this pattern I think it is comething we have come up against on previous occassions and then this workaround has been adopted. So I think a lot of developer time has already been spent on trying to fix this issue and I dont think it is that important an issue. 

The current test checks that on returm to the page we still see the banner. We can manually test that it works as part of UAT and I think we can assure ourselves that the functionality is there without having to cover it with an automated test.